libdevice/imf_bf16.hpp

```diff
@@ -81,6 +81,85 @@ __float2bfloat16(float f, __iml_rounding_mode rounding_mode) {
 
   return (bf16_sign << 15) | (bf16_exp << 7) | bf16_mant;
 }
+// We only need utils to convert double to bfloat16 with RTE
+static _iml_bf16_internal __double2bfloat16(double d) {
+  union {
+    double d_val;
+    uint64_t u64_val;
+  } fp64_bits;
+
+  fp64_bits.d_val = d;
+  uint16_t bf16_sign =
+      static_cast<uint16_t>((fp64_bits.u64_val & 0x8000000000000000) >> 63);
+  uint16_t fp64_exp =
+      static_cast<uint16_t>((fp64_bits.u64_val & 0x7FF0000000000000) >> 52);
+  uint64_t fp64_mant = (fp64_bits.u64_val & 0xFFFFFFFFFFFFF);
+  uint16_t bf16_mant;
+  // handling +/-infinity and NAN for double input
+  if (fp64_exp == 0x7FF) {
+    if (!fp64_mant) {
+      return bf16_sign ? 0xFF80 : 0x7F80;
+    } else {
+      // returns a quiet NaN
+      return 0x7FC0;
+    }
+  }
+
+  // Subnormal double precision is converted to 0
+  if (!fp64_exp) {
```

Comment:
i would put explicit `fp64_exp == 0` for better clarity.

Reply:
Done.
Thanks very much.